### PR TITLE
Handle HTTP errors and show in alert

### PR DESCRIPTION
**Pull request by i-schuetz**
Now HTTP error status are forwarded correctly as errors and shown in the alert + possible API error message.

### CODE REVIEW

File: CoEpi/api/CoEpiApi.swift [hunk 1]
```diff
@@ -33,36 +33,32 @@ class CoEpiApiImpl: CoEpiApi {
...
-                    emitter(.success(obj))
-                } catch let error {
-                    emitter(.error(ApiError.error(
-                        message: "Couldn't parse reponse: \(error), " +
-                            "data: \(String(describing: String(data: data, encoding: .utf8)))")))
+                if let apiError = response.apiError {
+                    emitter(.error(apiError))
+                } else {
+                    guard let data = response.data else {
+                        emitter(.error(ApiError.error(message: "Response didn't have data")))
```
Comment (duskoo) [inline CoEpi/api/CoEpiApi.swift:40]: Can "no response data" be a valid response?

Comment (i-schuetz) [reply CoEpi/api/CoEpiApi.swift:40]: Wouldn't expect it for a `get` request.

**Comment by i-schuetz** [comment]
@duskoo it would be a nice to have. I personally would recommend using Charles though (or another web proxy), it's nicer to use and has more functionality.

**Comment by i-schuetz** [comment]
@duskoo anyway add the interceptor if you want? I ended adding it to Android because Charles isn't working with the new endpoint (it works for iOS though).

**Pull request merged**